libsolidity/ast/TypeProvider.cpp

```diff
@@ -445,13 +445,18 @@ FunctionType const* TypeProvider::function(
 	strings const& _parameterTypes,
 	strings const& _returnParameterTypes,
 	FunctionType::Kind _kind,
-	bool _arbitraryParameters,
-	StateMutability _stateMutability
+	StateMutability _stateMutability,
+	FunctionType::Options _options
 )
 {
+	// Can only use this constructor for "arbitraryParameters".
+	solAssert(!_options.valueSet && !_options.gasSet && !_options.bound);
```

Comment:
Isn't salt missing here?
```suggestion
	solAssert(!_options.valueSet && !_options.gasSet && !options.saltSet && !_options.bound);
```